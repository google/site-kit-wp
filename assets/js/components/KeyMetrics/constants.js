/**
 * Key Metrics components - constants.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const KEY_METRICS_SETUP_CTA_WIDGET_SLUG = 'key-metrics-setup-cta-widget';

export const KEY_METRICS_SELECTION_PANEL_OPENED_KEY =
	'googlesitekit-key-metrics-selection-panel-opened';
export const KEY_METRICS_SELECTION_FORM = 'key-metrics-selection-form';
export const KEY_METRICS_SELECTED = 'key-metrics-selected';
export const EFFECTIVE_SELECTION = 'key-metrics-effective-selection';
export const UNSTAGED_SELECTION = 'key-metrics-unstaged-selection';
export const MIN_SELECTED_METRICS_COUNT = 2;
export const MAX_SELECTED_METRICS_COUNT = 4;
export const MAX_SELECTED_METRICS_COUNT_WITH_CONVERSION_EVENTS = 8;
export const KEY_METRICS_GROUP_CURRENT = {
	SLUG: 'current-selection',
	LABEL: __( 'Current selection', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_SUGGESTED = {
	SLUG: 'suggested',
	LABEL: __( 'Suggested', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_VISITORS = {
	SLUG: 'visitors',
	LABEL: __( 'Visitors', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_DRIVING_TRAFFIC = {
	SLUG: 'driving-traffic',
	LABEL: __( 'Driving traffic', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_GENERATING_LEADS = {
	SLUG: 'generating-leads',
	LABEL: __( 'Generating leads', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_SELLING_PRODUCTS = {
	SLUG: 'selling-products',
	LABEL: __( 'Selling products', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_CONTENT_PERFORMANCE = {
	SLUG: 'content-performance',
	LABEL: __( 'Content performance', 'google-site-kit' ),
};
