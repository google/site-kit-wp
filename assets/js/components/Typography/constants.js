/**
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const TYPE_DISPLAY = 'display';
export const TYPE_HEADLINE = 'headline';
export const TYPE_TITLE = 'title';
export const TYPE_BODY = 'body';
export const TYPE_LABEL = 'label';

export const VALID_TYPES = [
	TYPE_BODY,
	TYPE_DISPLAY,
	TYPE_HEADLINE,
	TYPE_LABEL,
	TYPE_TITLE,
];

export const SIZE_SMALL = 'small';
export const SIZE_MEDIUM = 'medium';
export const SIZE_LARGE = 'large';

export const VALID_SIZES = [ SIZE_SMALL, SIZE_MEDIUM, SIZE_LARGE ];
