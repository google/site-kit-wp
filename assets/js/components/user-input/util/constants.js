/**
 * User Input Answers.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const USER_INPUT_QUESTIONS_PURPOSE = 'purpose';
export const USER_INPUT_QUESTION_POST_FREQUENCY = 'postFrequency';
export const USER_INPUT_QUESTIONS_GOALS = 'goals';

export const USER_INPUT_QUESTIONS_LIST = [
	USER_INPUT_QUESTIONS_PURPOSE,
	USER_INPUT_QUESTION_POST_FREQUENCY,
	USER_INPUT_QUESTIONS_GOALS,
];

export const USER_INPUT_MAX_ANSWERS = {
	[ USER_INPUT_QUESTIONS_PURPOSE ]: 1,
	[ USER_INPUT_QUESTION_POST_FREQUENCY ]: 1,
	[ USER_INPUT_QUESTIONS_GOALS ]: 3,
};

export const USER_INPUT_CURRENTLY_EDITING_KEY =
	'googlesitekit-user-input-currently-editing';

export const FORM_USER_INPUT_QUESTION_NUMBER = 'user_input_question_number';
export const FORM_USER_INPUT_QUESTION_SNAPSHOT = 'user_input_question_snapshot';
export const USER_INPUT_LEGACY_SITE_PURPOSE_DISMISSED_ITEM_KEY =
	'user-input-legacy-site-purpose-dismissed-item';

/**
 * Gets available questions for user input settings.
 *
 * @since 1.139.0
 * @private
 *
 * @return {Object} Questions object.
 */
export function getUserInputQuestions() {
	return [
		{
			title: __(
				'What is the main purpose of this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answer, Site Kit will tailor the metrics you see on your dashboard to help you track progress towards your specific goals',
				'google-site-kit'
			),
		},
		{
			title: __(
				'How often do you create new content for this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answer, Site Kit will suggest new features for your dashboard related to content creation',
				'google-site-kit'
			),
		},
		{
			title: __(
				'What are your top goals for this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answers, Site Kit will tailor the metrics and recommendations you see on your dashboard to help you make progress in these areas',
				'google-site-kit'
			),
		},
	];
}

/**
 * Gets available answers for user input settings.
 *
 * @since 1.20.0
 * @private
 *
 * @return {Object} Answers object.
 */
export function getUserInputAnswers() {
	return {
		USER_INPUT_ANSWERS_PURPOSE: {
			sell_products_or_service: __(
				'Sell products or services',
				'google-site-kit'
			),
			sell_products: __( 'Sell products', 'google-site-kit' ),
			provide_services: __( 'Provide services', 'google-site-kit' ),
			monetize_content: __(
				'Monetize content (with ads or affiliate links)',
				'google-site-kit'
			),
			publish_blog: __( 'Publish a blog', 'google-site-kit' ),
			publish_news: __( 'Publish news content', 'google-site-kit' ),
			share_portfolio: __(
				'Share a business card or portfolio to represent me or my company online',
				'google-site-kit'
			),
			other: __( 'Other', 'google-site-kit' ),
		},
		USER_INPUT_ANSWERS_POST_FREQUENCY: {
			never: __( 'Never', 'google-site-kit' ),
			daily: __( 'Daily', 'google-site-kit' ),
			weekly: __( 'Weekly', 'google-site-kit' ),
			monthly: __( 'Monthly', 'google-site-kit' ),
			other: __( 'Other', 'google-site-kit' ),
		},
		USER_INPUT_ANSWERS_GOALS: {
			retaining_visitors: __(
				'Retain visitors, turn them into loyal readers or customers',
				'google-site-kit'
			),
			improving_performance: __(
				'Improve speed and performance',
				'google-site-kit'
			),
			finding_new_topics: __(
				'Find new topics to write about that connect with my audience',
				'google-site-kit'
			),
			growing_audience: __( 'Grow my audience', 'google-site-kit' ),
			expanding_business: __(
				'Expand my business into new cities, states or markets',
				'google-site-kit'
			),
			generating_revenue: __(
				'Generate more revenue',
				'google-site-kit'
			),
			help_better_rank: __(
				'Help my content rank in a better position in Google search results',
				'google-site-kit'
			),
			understanding_content_performance: __(
				'Understand which content is performing best',
				'google-site-kit'
			),
			encourage_to_post: __(
				'Tips for generating and posting engaging content updates',
				'google-site-kit'
			),
			other: __( 'Other', 'google-site-kit' ),
		},
	};
}

/**
 * Gets available answer descriptions for user input settings.
 *
 * @since 1.139.0
 * @private
 *
 * @return {Object} Answer descriptions object.
 */
export function getUserInputAnswersDescription() {
	return {
		USER_INPUT_ANSWERS_PURPOSE: {
			sell_products_or_service: __(
				'E.g. selling products like devices, apparel, equipment, etc. or offering services like courses, consulting, tutoring, etc.',
				'google-site-kit'
			),
			sell_products: __(
				'E.g. selling devices, apparel, equipment, etc.',
				'google-site-kit'
			),
			provide_services: __(
				'E.g. offering courses, consulting, tutoring, etc.',
				'google-site-kit'
			),
			monetize_content: __(
				'Using display ads, affiliate links, sponsored content, etc.',
				'google-site-kit'
			),
			publish_blog: __(
				'Writing on a topic youâ€™re passionate about, no focus on monetizing content',
				'google-site-kit'
			),
			publish_news: __(
				'E.g. local news, investigative pieces, interviews, etc.',
				'google-site-kit'
			),
			share_portfolio: __(
				'My website represents me or my company',
				'google-site-kit'
			),
			other: undefined,
		},
	};
}
