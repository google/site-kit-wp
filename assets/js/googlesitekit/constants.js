/**
 * Core constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const VIEW_CONTEXT_MAIN_DASHBOARD = 'mainDashboard';
export const VIEW_CONTEXT_ENTITY_DASHBOARD = 'entityDashboard';
export const VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY = 'mainDashboardViewOnly';
export const VIEW_CONTEXT_ENTITY_DASHBOARD_VIEW_ONLY =
	'entityDashboardViewOnly';
export const VIEW_CONTEXT_POSTS_LIST = 'postsList';
export const VIEW_CONTEXT_USER_INPUT = 'userInput';
export const VIEW_CONTEXT_ACTIVATION = 'activation';
export const VIEW_CONTEXT_SPLASH = 'splash';
export const VIEW_CONTEXT_ADMIN_BAR = 'adminBar';
export const VIEW_CONTEXT_ADMIN_BAR_VIEW_ONLY = 'adminBarViewOnly';
export const VIEW_CONTEXT_SETTINGS = 'settings';
export const VIEW_CONTEXT_AD_BLOCKING_RECOVERY = 'adBlockingRecovery';
export const VIEW_CONTEXT_MODULE = 'module';
export const VIEW_CONTEXT_WP_DASHBOARD = 'wpDashboard';
export const VIEW_CONTEXT_WP_DASHBOARD_VIEW_ONLY = 'wpDashboardViewOnly';
export const VIEW_CONTEXT_MODULE_SETUP = 'moduleSetup';
export const VIEW_CONTEXT_METRIC_SELECTION = 'metricSelection';

// Anchor ID Constants
export const ANCHOR_ID_KEY_METRICS = 'key-metrics';
export const ANCHOR_ID_TRAFFIC = 'traffic';
export const ANCHOR_ID_CONTENT = 'content';
export const ANCHOR_ID_SPEED = 'speed';
export const ANCHOR_ID_MONETIZATION = 'monetization';

// Site Kit only view contexts
export const SITE_KIT_VIEW_CONTEXTS = [
	VIEW_CONTEXT_MAIN_DASHBOARD,
	VIEW_CONTEXT_ENTITY_DASHBOARD,
	VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_ENTITY_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_USER_INPUT,
	VIEW_CONTEXT_SPLASH,
	VIEW_CONTEXT_SETTINGS,
	VIEW_CONTEXT_MODULE_SETUP,
	VIEW_CONTEXT_METRIC_SELECTION,
];

// Site Kit view-only contexts
export const SITE_KIT_VIEW_ONLY_CONTEXTS = [
	VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_ENTITY_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_ADMIN_BAR_VIEW_ONLY,
	VIEW_CONTEXT_WP_DASHBOARD_VIEW_ONLY,
];
