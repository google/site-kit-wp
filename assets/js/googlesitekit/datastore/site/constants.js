/**
 * `core/site` data store: constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { MODULE_SLUG_ADS } from '@/js/modules/ads/constants';
import { MODULE_SLUG_ANALYTICS_4 } from '@/js/modules/analytics-4/constants';
import { MODULE_SLUG_TAGMANAGER } from '@/js/modules/tagmanager/constants';

export const CORE_SITE = 'core/site';

export const AMP_MODE_PRIMARY = 'primary';
export const AMP_MODE_SECONDARY = 'secondary';

export const GOOGLE_TAG_GATEWAY_MODULES = [
	MODULE_SLUG_ANALYTICS_4,
	MODULE_SLUG_ADS,
	MODULE_SLUG_TAGMANAGER,
];
