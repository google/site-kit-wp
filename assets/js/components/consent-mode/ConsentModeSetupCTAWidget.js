/**
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';
import { useIntersection } from 'react-use';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import {
	Fragment,
	createInterpolateElement,
	useEffect,
	useRef,
	useState,
} from '@wordpress/element';

/**
 * Internal dependencies
 */
import { useSelect, useDispatch } from 'googlesitekit-data';
import { Button, SpinnerButton } from 'googlesitekit-components';
import { CORE_LOCATION } from '../../googlesitekit/datastore/location/constants';
import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
import { withWidgetComponentProps } from '../../googlesitekit/widgets/util';
import { Cell, Grid, Row } from '../../material-components';
import BannerGraphicsSVG from '../../../svg/graphics/consent-mode-setup.svg';
import BannerGraphicsTabletSVG from '../../../svg/graphics/consent-mode-setup-tablet.svg';
import {
	AdminMenuTooltip,
	useShowTooltip,
	useTooltipState,
} from '../AdminMenuTooltip';
import ErrorText from '../ErrorText';
import Link from '../Link';
import useViewContext from '../../hooks/useViewContext';
import useViewOnly from '../../hooks/useViewOnly';
import { DAY_IN_SECONDS, WEEK_IN_SECONDS, trackEvent } from '../../util';
import { CONSENT_MODE_SETUP_CTA_WIDGET_SLUG } from './constants';
import {
	BREAKPOINT_SMALL,
	BREAKPOINT_XLARGE,
	useBreakpoint,
} from '../../hooks/useBreakpoint';

function ConsentModeSetupCTAWidget( { Widget, WidgetNull } ) {
	const [ isSaving, setIsSaving ] = useState( false );
	const [ saveError, setSaveError ] = useState( null );

	const breakpoint = useBreakpoint();

	const viewContext = useViewContext();
	const viewOnlyDashboard = useViewOnly();

	const isConsentModeEnabled = useSelect( ( select ) =>
		select( CORE_SITE ).isConsentModeEnabled()
	);

	const settingsURL = useSelect( ( select ) =>
		select( CORE_SITE ).getAdminURL( 'googlesitekit-settings' )
	);

	const consentModeDocumentationURL = useSelect( ( select ) =>
		select( CORE_SITE ).getDocumentationLinkURL( 'consent-mode' )
	);

	const showTooltip = useShowTooltip( CONSENT_MODE_SETUP_CTA_WIDGET_SLUG );
	const { isTooltipVisible } = useTooltipState(
		CONSENT_MODE_SETUP_CTA_WIDGET_SLUG
	);

	const isDismissed = useSelect( ( select ) =>
		select( CORE_USER ).isPromptDismissed(
			CONSENT_MODE_SETUP_CTA_WIDGET_SLUG
		)
	);
	const dismissCount = useSelect( ( select ) =>
		select( CORE_USER ).getPromptDismissCount(
			CONSENT_MODE_SETUP_CTA_WIDGET_SLUG
		)
	);

	const usingProxy = useSelect( ( select ) =>
		select( CORE_SITE ).isUsingProxy()
	);

	const { setConsentModeEnabled, saveConsentModeSettings } =
		useDispatch( CORE_SITE );
	const { dismissPrompt, triggerSurvey } = useDispatch( CORE_USER );
	const { navigateTo } = useDispatch( CORE_LOCATION );

	const trackingRef = useRef();

	const intersectionEntry = useIntersection( trackingRef, {
		threshold: 0.25,
	} );
	const [ hasBeenInView, setHasBeenInView ] = useState( false );
	const inView = !! intersectionEntry?.intersectionRatio;

	const shouldShowWidget = useSelect( ( select ) => {
		if ( viewOnlyDashboard ) {
			return false;
		}

		if ( isSaving ) {
			return true;
		}

		if ( isDismissed !== false || isConsentModeEnabled !== false ) {
			return false;
		}

		return select( CORE_SITE ).isAdsConnected();
	} );

	useEffect( () => {
		if ( inView && ! hasBeenInView && shouldShowWidget ) {
			trackEvent(
				`${ viewContext }_CoMo-ads-setup-notification`,
				'view_notification'
			);

			if ( usingProxy ) {
				triggerSurvey( 'view_como_setup_cta', { ttl: DAY_IN_SECONDS } );
			}

			setHasBeenInView( true );
		}
	}, [
		hasBeenInView,
		inView,
		shouldShowWidget,
		triggerSurvey,
		usingProxy,
		viewContext,
	] );

	if ( isTooltipVisible ) {
		return (
			<Fragment>
				<WidgetNull />
				<AdminMenuTooltip
					title=""
					content={ __(
						'You can always enable consent mode from Settings later',
						'google-site-kit'
					) }
					dismissLabel={ __( 'Got it', 'google-site-kit' ) }
					tooltipStateKey={ CONSENT_MODE_SETUP_CTA_WIDGET_SLUG }
				/>
			</Fragment>
		);
	}

	if ( ! shouldShowWidget ) {
		return <WidgetNull />;
	}

	const handleCTAClick = async () => {
		setSaveError( null );
		setIsSaving( true );

		setConsentModeEnabled( true );

		const promises = [
			saveConsentModeSettings(),
			trackEvent(
				`${ viewContext }_CoMo-ads-setup-notification`,
				'confirm_notification'
			),
		];

		if ( usingProxy ) {
			promises.push(
				triggerSurvey( 'enable_como', { ttl: DAY_IN_SECONDS } )
			);
		}

		const [ { error } ] = await Promise.all( promises );

		if ( error ) {
			setSaveError( error );
			setConsentModeEnabled( false );
			setIsSaving( false );
		} else {
			await dismissPrompt( CONSENT_MODE_SETUP_CTA_WIDGET_SLUG );
			navigateTo( `${ settingsURL }#/admin-settings` );
		}
	};

	const handleDismissClick = async () => {
		trackEvent(
			`${ viewContext }_CoMo-ads-setup-notification`,
			'dismiss_notification'
		);

		showTooltip();

		// For the first two dismissals, we show the notification again in two weeks.
		if ( dismissCount < 2 ) {
			const twoWeeksInSeconds = WEEK_IN_SECONDS * 2;
			await dismissPrompt( CONSENT_MODE_SETUP_CTA_WIDGET_SLUG, {
				expiresInSeconds: twoWeeksInSeconds,
			} );
		} else {
			// For the third dismissal, dismiss permanently.
			await dismissPrompt( CONSENT_MODE_SETUP_CTA_WIDGET_SLUG );
		}
	};

	return (
		<div className="googlesitekit-widget-context">
			<Grid className="googlesitekit-widget-area">
				<Row>
					<Cell size={ 12 }>
						<Widget
							noPadding
							className="googlesitekit-setup-cta-banner googlesitekit-consent-mode-setup-cta-widget"
						>
							<div
								ref={ trackingRef }
								className="googlesitekit-setup-cta-banner__cells"
							>
								<div className="googlesitekit-setup-cta-banner__primary-cell">
									<h3 className="googlesitekit-setup-cta-banner__title">
										{ __(
											'Enable Consent Mode to preserve tracking for your Ads campaigns',
											'google-site-kit'
										) }
									</h3>
									<div className="googlesitekit-setup-cta-banner__description">
										<p>
											{ createInterpolateElement(
												__(
													'Consent mode interacts with your Consent Management Platform (CMP) or custom implementation for obtaining visitor consent, such as a cookie consent banner. <a>Learn more</a>',
													'google-site-kit'
												),
												{
													a: (
														<Link
															href={
																consentModeDocumentationURL
															}
															external
															aria-label={ __(
																'Learn more about consent mode',
																'google-site-kit'
															) }
														/>
													),
												}
											) }
										</p>
									</div>

									{ saveError && (
										<ErrorText
											message={ saveError.message }
										/>
									) }
									<div className="googlesitekit-setup-cta-banner__actions-wrapper">
										<Fragment>
											<SpinnerButton
												onClick={ handleCTAClick }
												isSaving={ isSaving }
											>
												{ __(
													'Enable consent mode',
													'google-site-kit'
												) }
											</SpinnerButton>
											<Button
												tertiary
												onClick={ handleDismissClick }
											>
												{ dismissCount < 2
													? __(
															'Maybe later',
															'google-site-kit'
													  )
													: __(
															'Donâ€™t show again',
															'google-site-kit'
													  ) }
											</Button>
										</Fragment>
									</div>
								</div>
								<div className="googlesitekit-setup-cta-banner__svg-wrapper">
									{ breakpoint !== BREAKPOINT_SMALL &&
									breakpoint !== BREAKPOINT_XLARGE ? (
										<BannerGraphicsTabletSVG />
									) : (
										<BannerGraphicsSVG />
									) }
								</div>
							</div>
						</Widget>
					</Cell>
				</Row>
			</Grid>
		</div>
	);
}

ConsentModeSetupCTAWidget.propTypes = {
	Widget: PropTypes.elementType.isRequired,
	WidgetNull: PropTypes.elementType.isRequired,
};

export default withWidgetComponentProps( 'consent-mode-setup-cta' )(
	ConsentModeSetupCTAWidget
);
