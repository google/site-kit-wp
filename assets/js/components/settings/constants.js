/**
 * Settings constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { MODULE_SLUG_ADS } from '../../modules/ads/datastore/constants';
import { MODULE_SLUG_SIGN_IN_WITH_GOOGLE } from '../../modules/sign-in-with-google/datastore/constants';
import { MODULE_SLUG_READER_REVENUE_MANAGER } from '../../modules/reader-revenue-manager/datastore/constants';

export const NEW_MODULES = [
	MODULE_SLUG_ADS,
	MODULE_SLUG_READER_REVENUE_MANAGER,
	MODULE_SLUG_SIGN_IN_WITH_GOOGLE,
];

export const BETA_MODULES = [ MODULE_SLUG_SIGN_IN_WITH_GOOGLE ];

export const EXPERIMENTAL_MODULES = [];
