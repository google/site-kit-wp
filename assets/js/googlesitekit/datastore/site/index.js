/**
 * `core/site` data store
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { combineStores, commonStore } from 'googlesitekit-data';
import cache from './cache';
import connection from './connection';
import consentMode from './consent-mode';
import conversionTracking from './conversion-tracking';
import errors from './errors';
import firstPartyMode from './first-party-mode';
import html from './html';
import info from './info';
import reset from './reset';
import enableAutoUpdate from './enable-auto-update';
import settings from './settings';
import urls from './urls';
import developerPlugin from './developer-plugin';
import { CORE_SITE } from './constants';
import notifications from './notifications';
import { createErrorStore } from '../../data/create-error-store';

const store = combineStores(
	commonStore,
	connection,
	consentMode,
	conversionTracking,
	errors,
	firstPartyMode,
	html,
	info,
	developerPlugin,
	reset,
	enableAutoUpdate,
	settings,
	urls,
	notifications,
	cache,
	createErrorStore( CORE_SITE )
);

export const initialState = store.initialState;
export const actions = store.actions;
export const controls = store.controls;
export const reducer = store.reducer;
export const resolvers = store.resolvers;
export const selectors = store.selectors;

export const registerStore = ( registry ) => {
	registry.registerStore( CORE_SITE, store );
};

export default store;
