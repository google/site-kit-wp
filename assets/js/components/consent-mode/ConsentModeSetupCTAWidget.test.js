/**
 * ConsentModeSetupCTAWidget component tests.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import {
	render,
	createTestRegistry,
	provideSiteInfo,
	provideUserInfo,
	provideUserAuthentication,
} from '../../../../tests/js/test-utils';
import ConsentModeSetupCTAWidget from './ConsentModeSetupCTAWidget';
import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
import { CONSENT_MODE_SETUP_CTA_WIDGET_SLUG } from './constants';

describe( 'ConsentModeSetupCTAWidget', () => {
	let registry;

	beforeEach( () => {
		registry = createTestRegistry();

		provideSiteInfo( registry );
		provideUserInfo( registry );
		provideUserAuthentication( registry );

		registry
			.dispatch( CORE_SITE )
			.receiveGetAdsMeasurementStatus( { connected: true } );

		registry.dispatch( CORE_SITE ).receiveGetConsentModeSettings( {
			enabled: false,
			regions: [ 'AT', 'EU' ],
		} );

		registry.dispatch( CORE_USER ).receiveGetDismissedPrompts( [] );
	} );

	it( 'should render the widget', async () => {
		const { container, waitForRegistry } = render(
			<ConsentModeSetupCTAWidget />,
			{
				registry,
			}
		);

		await waitForRegistry();

		expect( container ).toMatchSnapshot();
	} );

	it( 'should not render the widget when it is being dismissed', async () => {
		registry
			.dispatch( CORE_USER )
			.setIsPromptDimissing( CONSENT_MODE_SETUP_CTA_WIDGET_SLUG, true );

		const { container, waitForRegistry } = render(
			<ConsentModeSetupCTAWidget />,
			{
				registry,
			}
		);

		await waitForRegistry();

		expect( container ).toBeEmptyDOMElement();
	} );
} );
