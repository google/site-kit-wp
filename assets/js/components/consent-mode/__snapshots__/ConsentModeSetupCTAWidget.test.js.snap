// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsentModeSetupCTAWidget should render the widget 1`] = `
<div>
  <div
    class="googlesitekit-widget-context"
  >
    <div
      class="mdc-layout-grid googlesitekit-widget-area"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
        >
          <div
            class="googlesitekit-widget googlesitekit-widget--consent-mode-setup-cta googlesitekit-widget--no-padding googlesitekit-setup-cta-banner googlesitekit-consent-mode-setup-cta-widget"
          >
            <div
              class="googlesitekit-widget__body"
            >
              <div
                class="googlesitekit-setup-cta-banner__cells"
              >
                <div
                  class="googlesitekit-setup-cta-banner__primary-cell"
                >
                  <h3
                    class="googlesitekit-setup-cta-banner__title"
                  >
                    Enable Consent Mode to preserve tracking for your Ads campaigns
                  </h3>
                  <div
                    class="googlesitekit-setup-cta-banner__description"
                  >
                    <p>
                      Consent mode interacts with your Consent Management Platform (CMP) or custom implementation for obtaining visitor consent, such as a cookie consent banner. 
                      <a
                        aria-label="Learn more about consent mode (opens in a new tab)"
                        class="googlesitekit-cta-link"
                        href="https://sitekit.withgoogle.com/support/?doc=consent-mode"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        <span
                          class="googlesitekit-cta-link__contents"
                        >
                          Learn more
                        </span>
                        <span
                          class="googlesitekit-icon-wrapper"
                          style="margin-left: 5px;"
                        >
                          <svg />
                        </span>
                      </a>
                    </p>
                  </div>
                  <div
                    class="googlesitekit-setup-cta-banner__actions-wrapper"
                  >
                    <button
                      class="mdc-button googlesitekit-button-icon--spinner googlesitekit-button-icon--spinner__after mdc-button--raised"
                      target="_self"
                    >
                      <span
                        class="mdc-button__label"
                      >
                        Enable consent mode
                      </span>
                    </button>
                    <button
                      class="mdc-button mdc-button--tertiary"
                      target="_self"
                    >
                      <span
                        class="mdc-button__label"
                      >
                        Maybe later
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="googlesitekit-setup-cta-banner__svg-wrapper"
                >
                  <svg />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
